### ISSUE TITLE
[JIT / NNC] Incomplete Coverage of Activations

### ISSUE THREAD

**Issue by eellison** [pytorch/pytorch#56969]
## 🐛 Bug

We have run into internal and external models which are missing fusion opportunities because we a) don't support all activations in autodiff and b) don't support all activations in NNC. These are very easy to add and important for generating fusions. 

[List of Pointwise Activations](https://pytorch.org/docs/stable/nn.html#non-linear-activations-weighted-sum-nonlinearity):

Symbolic Script Support (necessary for autodiff):

- [ ] ELU
- [ ] Hardshrink
- [ ] Hardsigmoid
- [ ] Hardtanh
- [ ] Hardswish
- [ ] LeakyRelu
- [ ] LogSigmoid
- [ ] PReLU
- [x] ReLU
- [ ] ReLU6
- [ ] RReLU
- [ ] SELU
- [ ] CELU
- [ ] GELU
- [x] Sigmoid
- [ ] SiLU
- [ ] Softplus
- [ ] Softshrink
- [ ] Softsign
- [x] Tanh
- [ ] Tanhshrink
- [x] Threshold

NNC Support (necessary for fusion)

- [ ] ELU
- [ ] Hardshrink
- [ ] Hardsigmoid
- [x] Hardtanh
- [ ] Hardswish
- [ ] LeakyRelu
- [ ] LogSigmoid
- [ ] PReLU
- [x] ReLU
- [ ] ReLU6
- [ ] RReLU
- [ ] SELU
- [ ] CELU
- [ ] GELU
- [x] Sigmoid
- [ ] SiLU
- [ ] Softplus
- [ ] Softshrink
- [ ] Softsign
- [x ] Tanh
- [ ] Tanhshrink
- [x] Threshold

**Comment by eellison** [comment]
cc @desertfire hardsigmoid/hardtanh

**Comment by Krovatkin** [comment]
cc @Chillee @ZolotukhinM @eellison @penguinwu @navahgar @desertfire

**Comment by ZolotukhinM** [comment]
I've added non-activation ops to this issue as well, I don't think there is too much value in having a separate issue for them.

**Comment by eellison** [comment]
@ZolotukhinM sounds good, thanks! @Chillee do you mind commenting with a few activations you think are less common that we can leave off initial prioritization (maybe for OSS)

**Comment by Chillee** [comment]
@ZolotukhinM I marked some ones I already have that I can add.

@eellison I'm not intimately familiar with all of these (and I'm not sure if some of these are used internally), but `celu`, `tanhshrink`, and `hardshrink` stand out to me as not being used much.

**Comment by janeyx99** [comment]
TorchScript has been deprecated in favor of torch.export as of 2.11. https://pytorch.org/blog/pytorch-2-11-release-blog/ Closing this issue as we are not further developing on it--please ping if you believe this issue was erroneously closed.
